src/wbase/Task.cc

```diff
@@ -206,6 +217,13 @@ std::vector<Task::Ptr> Task::createTasks(std::shared_ptr<proto::TaskMsg> const&
     return vect;
 }
 
+string Task::getQueryString() const {
+    string qs = _userQueryInfo->getTemplate(_templateId);
+    boost::algorithm::replace_all(qs, CHUNK_TAG, to_string(_chunkId));
+    boost::algorithm::replace_all(qs, SUBCHUNK_TAG, to_string(_subchunkId));
+    return qs;
```

Comment:
Note that your current design of the class `Task` allows `_chunkId` and `_subchunkId` to be negative numbers. What kind of query those (negative) values would produce here? Perhaps you need to check if the task instance really has valid non-negative attributes here and throw an exception?

Reply:
It would produce an invalid query, if it ever got that far, and would then trip a user error. The chunk/subchunk numbers tend to show up in the middle of table names  "Object_-1sub" will kick back a SQL error, which is better than "Object_0sub", which could result in invalid rows being added.

I think it's better in this case to generate the SQL error as this should not happen. CHUNK_TAG and SUBCHUNK_TAG should never be found in a query that does not have chunk or subchunk numbers in them.

Reply:
It's better to throw an exception in the ctor of the class `Task` and not wait (potentially for hours) before the query generator will be called when the task would be allowed to run for real. The code that does an early parameter validation in general is much safer than this one.

Reply:
The test is already there. Why should I test for the same thing again? 
From `void SsiRequest::execute(XrdSsiRequest& req)` before it tries making Tasks.
```
          if (!taskMsg->has_db() || !taskMsg->has_chunkid() || (ru.db() != taskMsg->db()) ||
                (ru.chunk() != taskMsg->chunkid())) {
                reportError("Mismatched db/chunk in TaskMsg on resource db=" + ru.db() +
                            " chunkId=" + std::to_string(ru.chunk()));
                return;
            }
```

There is no easy test for a valid subchunkId or if a query should even have one, but those checks are being made.

Reply:
That's good.